packages/models-library/src/models_library/utils/docker_compose.py

```diff
@@ -24,16 +27,15 @@ def replace_env_vars_in_compose_spec(
     returns a stringified version
     """
 
-    stringified_service_spec = yaml.safe_dump(service_spec)
+    content: str = yaml.safe_dump(service_spec)
 
-    # NOTE: could not use `string.Template` here because the test will
-    # fail since `${DISPLAY}` cannot be replaced, and we do not want
-    # it to be replaced at this time. If this method is changed
-    # the test suite should always pass without changes.
-    stringified_service_spec = stringified_service_spec.replace(
-        MATCH_SIMCORE_REGISTRY, replace_simcore_registry
+    template = TemplateText(content)
+    substitutions = SubstitutionsDict(
+        {
+            "SERVICE_VERSION": replace_service_version,
```

Comment:
Please format `MATCH_SERVICE_VERSION` via the same function you are using to create the replacement names, instead of using the string `SERVICE_VERSION` directly.